File: synara/src/app/features/matrix-dto/ids.ts
/**
 * Opaque Matrix identifier string aliases (product strings, not SDK types).
 */

/** Room id, e.g. `!opaque:example.org`. */
export type RoomId = string;

/** User id, e.g. `@alice:example.org`. */
export type UserId = string;

/** Event id, e.g. `$opaqueEventId`. */
export type EventId = string;

/** Device id, e.g. `DEVICEID`. */
export type DeviceId = string;

/** Opaque product media handle id (not an mxc URI). */
export type MediaHandleId = string;

/** Opaque upload job id. */
export type UploadId = string;

/** Opaque notification candidate id. */
export type NotificationCandidateId = string;

/** Opaque widget / call session id. */
export type WidgetId = string;

/** Timeline virtualization key. */
export type TimelineItemId = string;

File: src-tauri/src/matrix/dto/ids.rs
//! Opaque Matrix identifier string types (product strings, not Ruma types).
//!
//! These are plain `String` aliases with documentation. Validation beyond
//! non-empty checks belongs at the host boundary in later phases.

/// Room id, e.g. `!opaque:example.org`.
pub type RoomId = String;

/// User id, e.g. `@alice:example.org`.
pub type UserId = String;

/// Event id, e.g. `$opaqueEventId`.
pub type EventId = String;

/// Device id, e.g. `DEVICEID`.
pub type DeviceId = String;

/// Opaque product media handle id (not an mxc URI).
pub type MediaHandleId = String;

/// Opaque upload job id.
pub type UploadId = String;

/// Opaque notification candidate id.
pub type NotificationCandidateId = String;

/// Timeline virtualization key (often equals `eventId` or a local-echo id).
pub type TimelineItemId = String;
